backend/src/main/java/ai/verta/modeldb/versioning/blob/diff/ConflictGenerator.java

```diff
@@ -397,23 +400,28 @@ private static AutogenCodeDiff getCodeConflictBlob(
 
   private static AutogenNotebookCodeDiff getNoteBookDiff(
       AutogenNotebookCodeDiff a, AutogenNotebookCodeDiff b, AutogenNotebookCodeBlob c) {
-    return Utils.removeEmpty(
+
+    AutogenNotebookCodeDiff diff =
         new AutogenNotebookCodeDiff()
             .setGitRepo(
                 getGitDiff(
                     a == null ? null : a.getGitRepo(),
                     b == null ? null : b.getGitRepo(),
-                    c == null ? null : c.getGitRepo()))
-            .setPath(
-                new AutogenPathDatasetComponentDiff()
-                    .setA(a == null || a.getPath() == null ? null : a.getPath().getB())
-                    .setB(b == null || b.getPath() == null ? null : b.getPath().getB())
-                    .setC(c == null ? null : c.getPath())
-                    .setStatus(AutogenDiffStatusEnumDiffStatus.fromProto(DiffStatus.CONFLICTED))));
+                    c == null ? null : c.getGitRepo()));
+    if (((a == null || a.getPath() == null) && (b == null || b.getPath() == null))
+        || (a != null && a.equals(b))) return Utils.removeEmpty(diff);
+    diff.setPath(
+        new AutogenPathDatasetComponentDiff()
+            .setA(a == null || a.getPath() == null ? null : a.getPath().getB())
```

Comment:
```suggestion
            .setA(a == null || a.getPath() == null ? null : a.getPath().getA())
```
Maybe?

Reply:
no these are the diffs of the commits being merged so get B is correct , getA would be the common parent

Reply:
Sg